pkg/actuators/machine/instances.go

```diff
@@ -422,9 +427,29 @@ func buildTagList(machineName string, clusterID string, machineTags []awsprovide
 		{Key: aws.String("kubernetes.io/cluster/" + clusterID), Value: aws.String("owned")},
 		{Key: aws.String("Name"), Value: aws.String(machineName)},
 	}...)
+
 	return removeDuplicatedTags(rawTagList)
 }
 
+// mergeInfrastructureAndMachineSpecTags merge list of tags from machine provider spec and Infrastructure object platform spec.
+// Machine tags have precedence over Infrastructure
+func mergeInfrastructureAndMachineSpecTags(machineSpecTags []awsproviderv1.TagSpecification, infra *configv1.Infrastructure) []awsproviderv1.TagSpecification {
+	if infra == nil || infra.Status.PlatformStatus == nil || infra.Status.PlatformStatus.AWS == nil || infra.Status.PlatformStatus.AWS.ResourceTags == nil {
+		return machineSpecTags
+	}
+
+	mergedList := []awsproviderv1.TagSpecification{}
+	mergedList = append(mergedList, machineSpecTags...)
+
+	for _, tag := range infra.Status.PlatformStatus.AWS.ResourceTags {
+		mergedList = append(mergedList, awsproviderv1.TagSpecification{Name: tag.Key, Value: tag.Value})
+	}
+
+	machineSpecTags = append(machineSpecTags, mergedList...)
+
+	return machineSpecTags
+}
```

Comment:
We currently have this duplicated now
```suggestion
	mergedList := []awsproviderv1.TagSpecification{}
	mergedList = append(mergedList, machineSpecTags...)

	for _, tag := range infra.Status.PlatformStatus.AWS.ResourceTags {
		mergedList = append(mergedList, awsproviderv1.TagSpecification{Name: tag.Key, Value: tag.Value})
	}

	return mergedList
}
```

Reply:
🤦‍♂️